.github/workflows/ci.yaml

```diff
@@ -0,0 +1,134 @@
+# Add a comment to force a change.
+
+name: Webhook CI
+
+on:
+  workflow_call:
+  push:
+    branches:
+      - release/v*
+    paths-ignore:
+      - '*.md'
+      - '.gitignore'
+      - 'CODEOWNERS'
+      - 'LICENSE'
+  pull_request:
+    paths-ignore:
+    - '*.md'
+    - '.gitignore'
+    - 'CODEOWNERS'
+    - 'LICENSE'
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    name: CI
+    runs-on : ubuntu-latest
+    strategy:
+      matrix:
+          arch:
+          - amd64
+          - arm64
+    steps:
+      - name : Checkout repository
+        uses : actions/checkout@v4
+        with :
+          fetch-depth : 0
+      - run : make ci
+        name: ci
+
+      - name : Save builds
+        run : |
+          mkdir -p dist/artifacts
+          cp bin/* dist/artifacts/
+      - name: Upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: webhook-build-artifacts-${{ matrix.arch }}
+          path: ./dist/artifacts
+
+  test:
+    name : integration-test
+    needs : [
+      build
+    ]
+    runs-on : ubuntu-latest
+    strategy:
+      matrix:
+        arch:
+        - amd64
+        - arm64
+    steps:
+      - name : Checkout repository
+        uses : actions/checkout@v4
+        with :
+          fetch-depth : 0
+
+      # TODO: Pull this next one out once there's a release of rancher 2.9 that runs on k8s 1.29
+      - name : Checkout rancher/rancher and build the chart
+        run: |
+          pushd "${{ runner.temp}}"
+          git clone https://github.com/rancher/rancher.git rancherDir
+          cd rancherDir
+          git checkout release/v2.9
```

Comment:
This could be replaced by 

```sh
git clone -b release/v2.9 https://github.com/rancher/rancher.git rancherDir
cd rancherDir
```

and perhaps even the following to avoid downloading the whole history for that repo.

```sh
git clone --depth 1 -b release/v2.9 https://github.com/rancher/rancher.git rancherDir
cd rancherDir
```

We could also just re-use the `actions/checkout@v4` action for this. Something like:

```sh
- uses: actions/checkout@v4
  with:
    repository: 'rancher/rancher'
    ref: 'release/v2.9'
    path: 'rancherDir'
```

Reply:
And if I pull  out the part of this that supports k8s 1.29, none of this is needed.